net/src/eth/mod.rs

```diff
@@ -68,106 +49,57 @@ impl Eth {
     /// # Errors
     ///
     /// This function will return an error if the specified source or dest [Mac] are invalid.
-    pub fn new(source: Mac, destination: Mac, ether_type: EthType) -> Result<Eth, EthError> {
-        let mut header = Eth(Ethernet2Header {
-            source: source.0,
-            destination: destination.0,
+    pub fn new(
```

Comment:
I'm a bit confused here. According to the documentation line, this function may return an error. However, it seems to unconditionally return Ok(), without any of the member initialization being able to fail.

Reply:
yeah, this is the result of an automated refactor.  Good catch